interfaces/sourcegen/sourcegen/_TagFileParser.py

```diff
@@ -214,13 +222,20 @@ def tag_info(self, func_string: str) -> TagInfo:
                     _LOGGER.critical(msg)
                     sys.exit(1)
 
-        return TagInfo.from_xml(cxx_func, self._known[cxx_func][ix])
+        return TagInfo.from_xml(cxx_member, self._known[cxx_member][ix])
 
-    def cxx_func(self, func_string: str) -> CFunc:
-        """Generate annotated C++ function specification."""
+    def cxx_member(self, func_string: str, setter: bool = False) -> CFunc | Param:
+        """Generate annotated C++ function/variable specification."""
         details = tag_lookup(self.tag_info(func_string))
         ret_param = Param.from_xml(details.type)
 
+        if details.kind == "variable":
+            if setter:
+                return Param(ret_param.p_type, details.name,
+                             details.briefdescription, "in", None, details.base)
+            return Param(ret_param.p_type, details.name,
+                         details.briefdescription, "out", None, details.base)
```

Comment:
This could be slightly less redundant if you just define something like `direction = "in" if setter else "out"` and pass that as an argument to `Param`.